go-controller/pkg/cni/ovs.go

```diff
@@ -261,6 +315,10 @@ func waitForPodInterface(ctx context.Context, mac string, ifAddrs []*net.IPNet,
 				}
 			}
 
+			if err := checkCancelSandbox(mac, podLister, kclient, namespace, name, initialPodUID); err != nil {
+				return fmt.Errorf("%v waiting for OVS port binding for %s %v", err, mac, ifAddrs)
+			}
+
 			// try again later
```

Comment:
Why do we try again later? Shouldn't this just fail gracefully and then we will get another CNI ADD event at some point?

Reply:
@astoycos good question; the function really has two purposes: (a) wait up to 20s for ovn-controller to install flows and (b) stop waiting if we know the sandbox is no longer relevant. So the "try again later" bit is what continues waiting for OVN to install the flows, and everything above it is for the "is this SB still relevant" part.